docs/testing.rst

```diff
@@ -0,0 +1,78 @@
+Testing Services
+================
+
+Philosophy
+----------
+
+Nameko's conventions are designed to make testing as easy as possible. Services are likely to be small and single-purpose, and dependency injection makes it simple to replace and isolate pieces of functionality.
+
+Unit Testing
+------------
+
+Unit testing in nameko usually means testing a single service in isolation - i.e. without any or most of its dependencies.
+
+Tne :func:`~nameko.testing.services.worker_factory` utility will create a worker from a given service class, with its dependencies replaced by :class:`mock.MagicMock` objects. Dependency functionality can then be imitated by adding side effects and return values:
```

Comment:
mention that examples use pytest(link) which is what nameko's own test suite is using, but all testing utilities provided are test framework agnostic